olive/evaluator/olive_evaluator.py

```diff
@@ -3,37 +3,421 @@
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
 import logging
-from typing import Dict, List
+import time
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
 
+import numpy as np
+import torch
 from pydantic import validator
+from torch.utils.data import Dataset
 
 from olive.common.config_utils import ConfigBase
-from olive.evaluator.metric import Metric
-from olive.model import OliveModel
-from olive.systems.local import LocalSystem
-from olive.systems.olive_system import OliveSystem
+from olive.common.user_module_loader import UserModuleLoader
+from olive.common.utils import tensor_data_to_device
+from olive.constants import Framework
+from olive.evaluator.accuracy import AUC, AccuracyScore, F1Score, Precision, Recall
+from olive.evaluator.metric import AccuracySubType, LatencySubType, Metric, MetricType
+from olive.model import OliveModel, ONNXModel, OpenVINOModel, PyTorchModel, SNPEModel
+from olive.systems.common import Device
 
 logger = logging.getLogger(__name__)
 
 
-class OliveEvaluator:
-    def __init__(self, metrics: List[Metric]):
-        metric_names = set([metric.name for metric in metrics])
-        assert len(metric_names) == len(metrics), "Metric names must be unique"
-        self.metrics = metrics
+class DummyDataloader(Dataset):
```

Comment:
Can you please explain me a little bit in which case we need this DummyDataloader? Do we still need this since we have an olive data container already?

Reply:
This is carry -forward from previous implementation. If it's no more needed, then it should be removed in it's own PR.

Reply:
Looks like we still tests like bert_nic_ptq that provide inputs using the old input_names, input_shapes, etc which would need DummyDataloader to pass.

Reply:
We can double check this and remove this if this is no need for this any more. @trajepl do you know if this is needed?